central/debug/service/service.go

```diff
@@ -214,6 +234,32 @@ func (s *serviceImpl) StreamAuthzTraces(_ *v1.Empty, stream v1.DebugService_Stre
 	}
 }
 
+func getJSONToZipWithLogging(zipWriter *zip.Writer, fileName string) func(interface{}, error) {
+	return func(jsonObj interface{}, err error) {
+		if err != nil {
```

Comment:
This looks weird to me. How about accepting function that returns `jsonObj interface{}, err error` so we have call and error check close to each other?